sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -225,6 +225,11 @@ case class AlterTableSetPropertiesCommand(
     val catalog = sparkSession.sessionState.catalog
     val table = catalog.getTableMetadata(tableName)
     DDLUtils.verifyAlterTableType(catalog, table, isView)
+    // Not allowed to switch the table type.
+    if (properties.contains("EXTERNAL")) {
```

Comment:
This is officially documented in the Hive document, as shown in the [link](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL):
`TBLPROPERTIES ("EXTERNAL"="TRUE") in release 0.6.0+ (HIVE-1329) – Change a managed table to an external table and vice versa for "FALSE".`

This is the only property users are not allowed to change. The other Hive-specific properties are still allowed to change, because Hive also allows it. 

For the our Spark-reserved properties, users are not allowed to change. See the function call `verifyTableProperties` in `[alterTable](https://github.com/apache/spark/blob/b9a147181d5e38d9abed0c7215f4c5cb695f579c/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala#L393)`.